lib/libc/string/lib_strdup.c

```diff
@@ -67,10 +67,11 @@
 FAR char *strdup(const char *s)
 {
 	FAR char *news = NULL;
+	size_t s_len = strlen(s) + 1;
 	if (s) {
-		news = (FAR char *)lib_malloc(strlen(s) + 1);
+		news = (FAR char *)lib_malloc(s_len);
 		if (news) {
-			strcpy(news, s);
+			strncpy(news, s, s_len);
```

Comment:
If `s` has no `\0` char,  `strlen` makes assert.
If `s` has `\0`, `strdup` works well.
I think we don't have to use `strncpy`